testfixtures/logcapture.py

```diff
@@ -73,11 +86,53 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        return self._rec_2_obj(self.records[key])
+
+    def __contains__(self, what):
+        for i, item in enumerate(self):
+            if what == item:
+                self.records[i].is_asserted = True
+                return True
+        return False
+
     def clear(self):
         """Clear any entries that have been captured."""
         self.records = []
 
+    def mark_logs_as_asserted(self):
+        """ Mark all captured logs as asserted,
+         use this when doing the assertion externally """
```

Comment:
nitpick: please have the triple quotes on their own lines to match the style in the rest of this codebase.